Interesting that you're not reproducing.  I suppose it's possible it's something weird about my environment.  Here's my exact steps, with output:

```
$ git diff
diff --git a/Cargo.toml b/Cargo.toml
index 872da6b..285a980 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,7 +17,7 @@ include = ["src/*", "LICENSE-MIT.md", "README.md"]
 document-features = { version = "0.2", optional = true }
 encoding_rs = { version = "0.8", optional = true }
 serde = { version = "1.0.100", optional = true }
-tokio = { version = "1.0", optional = true, default-features = false, features = ["io-util"] }
+#tokio = { version = "1.0", optional = true, default-features = false, features = ["io-util"] }
 memchr = "2.0"

 [dev-dependencies]
@@ -49,7 +49,7 @@ default = []
 ## [reading events] from types implementing [`tokio::io::AsyncBufRead`].
 ##
 ## [reading events]: crate::reader::Reader::read_event_into_async
-async-tokio = ["tokio"]
+#async-tokio = ["tokio"]

 ## Enables support of non-UTF-8 encoded documents. Encoding will be inferred from
 ## the XML declaration if it will be found, otherwise UTF-8 is assumed.
```

```
$ cargo minimal-versions check
info: running `rustup run nightly cargo update -Z minimal-versions`
    Updating crates.io index
    Removing aho-corasick v0.7.20
    Removing anes v0.1.6
    Removing async-stream v0.3.4
    Removing async-stream-impl v0.3.4
    Removing atty v0.2.14
    Removing autocfg v1.1.0
    Removing bitflags v1.3.2
    Removing bumpalo v3.12.0
    Removing bytes v1.4.0
    Removing cast v0.3.0
    Updating cfg-if v1.0.0 -> v0.1.0
    Removing ciborium v0.2.0
    Removing ciborium-io v0.2.0
    Removing ciborium-ll v0.2.0
    Removing clap v3.2.23
    Removing clap_lex v0.2.4
    Removing criterion v0.4.0
    Removing criterion-plot v0.5.0
    Removing crossbeam-channel v0.5.6
    Removing crossbeam-deque v0.8.2
    Removing crossbeam-epoch v0.9.13
    Removing crossbeam-utils v0.8.14
    Removing ctor v0.1.26
    Removing diff v0.1.13
    Updating document-features v0.2.7 -> v0.2.0
    Removing either v1.8.1
    Updating encoding_rs v0.8.32 -> v0.8.0
    Removing futures-core v0.3.26
    Removing half v1.8.2
    Removing hashbrown v0.12.3
    Removing hermit-abi v0.1.19
    Removing hermit-abi v0.2.6
    Removing indexmap v1.9.2
    Removing itertools v0.10.5
    Removing itoa v1.0.5
    Removing js-sys v0.3.61
    Removing lazy_static v1.4.0
    Updating libc v0.2.139 -> v0.2.18
    Removing litrs v0.2.3
    Removing log v0.4.17
    Updating memchr v2.5.0 -> v2.0.0
    Removing memoffset v0.7.1
    Removing num-traits v0.2.15
    Removing num_cpus v1.15.0
    Removing once_cell v1.17.1
    Removing oorandom v11.1.3
    Removing ordered-float v2.10.0
    Removing os_str_bytes v6.4.1
    Removing output_vt100 v0.1.3
    Removing pin-project-lite v0.2.9
    Removing plotters v0.3.4
    Removing plotters-backend v0.3.4
    Removing plotters-svg v0.3.3
    Removing pretty_assertions v1.3.0
    Removing proc-macro2 v1.0.51
    Updating quote v1.0.23 -> v0.3.8
    Removing rayon v1.6.1
    Removing rayon-core v1.10.2
    Removing regex v1.7.1
    Removing regex-syntax v0.6.28
    Removing ryu v1.0.12
    Removing same-file v1.0.6
    Removing scopeguard v1.1.0
    Updating serde v1.0.152 -> v1.0.100
    Removing serde-value v0.7.0
    Updating serde_derive v1.0.152 -> v1.0.0
      Adding serde_derive_internals v0.15.0
    Removing serde_json v1.0.93
    Updating syn v1.0.108 -> v0.11.10
      Adding synom v0.11.0
    Removing textwrap v0.16.0
    Removing tinytemplate v1.2.1
    Removing tokio v1.25.0
    Removing tokio-macros v1.8.2
    Removing tokio-stream v0.1.12
    Removing tokio-test v0.4.2
    Removing unicode-ident v1.0.6
      Adding unicode-xid v0.0.4
    Removing walkdir v2.3.2
    Removing wasm-bindgen v0.2.84
    Removing wasm-bindgen-backend v0.2.84
    Removing wasm-bindgen-macro v0.2.84
    Removing wasm-bindgen-macro-support v0.2.84
    Removing wasm-bindgen-shared v0.2.84
    Removing web-sys v0.3.61
    Removing winapi v0.3.9
    Removing winapi-i686-pc-windows-gnu v0.4.0
    Removing winapi-util v0.1.5
    Removing winapi-x86_64-pc-windows-gnu v0.4.0
    Removing windows-sys v0.42.0
    Removing windows_aarch64_gnullvm v0.42.1
    Removing windows_aarch64_msvc v0.42.1
    Removing windows_i686_gnu v0.42.1
    Removing windows_i686_msvc v0.42.1
    Removing windows_x86_64_gnu v0.42.1
    Removing windows_x86_64_gnullvm v0.42.1
    Removing windows_x86_64_msvc v0.42.1
    Removing yansi v0.5.1
info: running `cargo hack check`
info: running `cargo check` on quick-xml (1/1)
    Checking quick-xml v0.27.1 (/home/dburgener/quick-xml)
error[E0432]: unresolved import `memchr::memchr2_iter`
 --> src/escapei.rs:3:5
  |
3 | use memchr::memchr2_iter;
  |     ^^^^^^^^^^^^^^^^^^^^ no `memchr2_iter` in the root

error[E0425]: cannot find function `memchr_iter` in crate `memchr`
  --> src/reader/parser.rs:96:46
   |
96 |                     if let Some(p) = memchr::memchr_iter(b'-', &buf[3..len - 2])
   |                                              ^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr_iter` in crate `memchr`
   --> src/reader/mod.rs:773:26
    |
773 |         for i in memchr::memchr_iter(b'>', chunk) {
    |                          ^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr2_iter` in crate `memchr`
   --> src/reader/mod.rs:815:43
    |
815 |                     let balance = memchr::memchr2_iter(b'<', b'>', content)
    |                                           ^^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr3_iter` in crate `memchr`
   --> src/reader/mod.rs:853:26
    |
853 |         for i in memchr::memchr3_iter(b'>', b'\'', b'"', chunk) {
    |                          ^^^^^^^^^^^^ not found in `memchr`

Some errors have detailed explanations: E0425, E0432.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `quick-xml` due to 5 previous errors
error: process didn't exit successfully: `/home/dburgener/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo check --manifest-path Cargo.toml` (exit status: 101)
error: process didn't exit successfully: `/home/dburgener/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo hack check` (exit status: 1)
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -18,7 +18,7 @@ document-features = { version = "0.2", optional = true }
 encoding_rs = { version = "0.8", optional = true }
 serde = { version = "1.0.100", optional = true }
 tokio = { version = "1.0", optional = true, default-features = false, features = ["io-util"] }
-memchr = "^2.1"
+memchr = "2.1"
 
 [dev-dependencies]
 criterion = "0.4"

--- a/Changelog.md
+++ b/Changelog.md
@@ -32,6 +32,7 @@
   `DeEvent::CData` events was replaced by `DeEvent::Text` with merged content.
   The same behavior for the `Reader` does not implemented (yet?) and should be
   implemented manually
+- [#562]: Correctly set minimum required version of memchr dependency to 2.1
 
 ### Misc Changes
 

--- a/Changelog.md
+++ b/Changelog.md
@@ -44,6 +44,7 @@
 [#540]: https://github.com/tafia/quick-xml/issues/540
 [#541]: https://github.com/tafia/quick-xml/pull/541
 [#556]: https://github.com/tafia/quick-xml/pull/556
+[#562]: https://github.com/tafia/quick-xml/pull/562
 
 ## 0.27.1 -- 2022-12-28
 

